src/function/scalar/operators/add.cpp

```diff
@@ -44,11 +44,18 @@ interval_t AddOperator::Operation(interval_t left, interval_t right) {
 
 template <>
 date_t AddOperator::Operation(date_t left, int32_t right) {
-	int32_t result;
-	if (!TryAddOperator::Operation(left.days, right, result)) {
+	if (!Value::IsFinite(left)) {
```

Comment:
So this means that `left` is either `infinity` or `-infinity`? In which case any addition would also return `infinity` or `-infinity`?

Reply:
Yes. (That is what PG does).